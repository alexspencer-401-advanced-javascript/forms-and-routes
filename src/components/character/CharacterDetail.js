import React from 'react';
import PropTypes from 'prop-types';

const CharacterDetail = ({ id, name, photoUrl, gender, hair, affiliation, weapon }) => {

  return (
    <div key={id}>
      <figure>
        <img src={photoUrl} alt={name} />
        <figcaption>{name}</figcaption>
      </figure>
      <p>{gender}</p>
      <p>{hair}</p>
      <p>{affiliation}</p>
      <p>{weapon}</p>
    </div>
  );
};

CharacterDetail.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  photoUrl: PropTypes.string,
  gender: PropTypes.string,
  hair: PropTypes.string,
  affiliation: PropTypes.string,
  weapon: PropTypes.string
};

export default CharacterDetail;
