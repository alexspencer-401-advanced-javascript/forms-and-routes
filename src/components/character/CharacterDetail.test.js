import React from 'react';
import { shallow } from 'enzyme';
import CharacterDetail from './CharacterDetail';

describe('CharacterDetail component', () => {
  it('renders CharacterDetail', () => {
    const wrapper = shallow(<CharacterDetail />);
    expect(wrapper).toMatchSnapshot();
  });
});
