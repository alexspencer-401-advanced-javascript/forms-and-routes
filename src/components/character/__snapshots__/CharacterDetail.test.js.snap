// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetail component renders CharacterDetail 1`] = `
<div>
  <figure>
    <img />
    <figcaption />
  </figure>
  <p />
  <p />
  <p />
  <p />
</div>
`;
